packages/dd-trace/src/tagger.js

```diff
@@ -5,25 +5,44 @@ const log = require('./log')
 const ERROR_MESSAGE = constants.ERROR_MESSAGE
 const ERROR_STACK = constants.ERROR_STACK
 const ERROR_TYPE = constants.ERROR_TYPE
+const DD_EMPTY_USER_TAG = constants.DD_EMPTY_USER_TAG
 
 const otelTagMap = {
   'deployment.environment': 'env',
   'service.name': 'service',
   'service.version': 'version'
 }
 
-function add (carrier, keyValuePairs, parseOtelTags = false) {
+// don't allow these tags to be empty
+const skipEmptyString = new Set(['env', 'service', 'version'])
+
+function add (carrier, keyValuePairs, parseOtelTags = false, parseSpaceSeparatedTags = false) {
   if (!carrier || !keyValuePairs) return
 
   if (Array.isArray(keyValuePairs)) {
-    return keyValuePairs.forEach(tags => add(carrier, tags))
+    return keyValuePairs.forEach(tags => add(carrier, tags, parseOtelTags, parseSpaceSeparatedTags))
   }
   try {
     if (typeof keyValuePairs === 'string') {
-      const segments = keyValuePairs.split(',')
+      let segments
+      if (parseSpaceSeparatedTags) {
+        const separator = keyValuePairs.includes(',') ? ',' : ' '
+        segments =
+        separator === ' '
+          ? keyValuePairs.split(/\s+/)
+          : keyValuePairs.split(separator)
+      } else {
+        segments = keyValuePairs.split(',')
+      }
+
       for (const segment of segments) {
         const separatorIndex = parseOtelTags ? segment.indexOf('=') : segment.indexOf(':')
-        if (separatorIndex === -1) continue
+        if (separatorIndex === -1) {
+          if (parseSpaceSeparatedTags) {
```

Comment:
It would probably be more efficient to have an entirely different code path when this option is on. Benchmarks would show us that though (although it's worth noting that benchmarks don't always show all the possible decisions that v8 can take for a piece of code)

Reply:
It would be difficult to implement a completely separate code path without duplicating a significant amount of the existing code

Reply:
Yeah that is definitely not ideal, but it's better speed-wise to avoid going through unneeded code, even if it means duplication. However, if the benchmarks don't show any significant slowdown it would definitely be fine to keep it all together. Basically I'm not blocking on separating, it's just one possible way to tackle the problem.

This type of optimization is something @BridgeAR excels at, so pinging him as well if he also wants to take a look, especially since tagger is called in pretty much the hottest code path of the library (multiple times for every span).

Reply:
Turns out space support is only needed when the environment variable is read, so the tagger doesn't need to be changed, support can be added only in the config class.